src/CabalHelper/Compiletime/Compile.hs

```diff
@@ -134,13 +134,14 @@ compileHelper' CompHelperEnv {..} = do
             ]
   appdir <- appCacheDir
   let cp@CompPaths {compExePath} = compPaths appdir cheProjLocalCacheDir comp
-  helper_exists <- doesFileExist compExePath
+  let compExePathExt = compExePath ++ exeExtension
```

Comment:
[the docs](https://hackage.haskell.org/package/directory-1.3.6.1/docs/System-Directory.html#v:exeExtension) says it includes the dot (and it will not work in other case), but as `<.>` adds it only if necessary, we can use it